unity-environment/Assets/ML-Agents/Examples/BananaCollectors/Scripts/BananaAgent.cs

```diff
@@ -42,15 +43,18 @@ public override void InitializeAgent()
 
     public override void CollectObservations()
     {
-        float rayDistance = 50f;
-        float[] rayAngles = { 20f, 90f, 160f, 45f, 135f, 70f, 110f };
-        string[] detectableObjects = { "banana", "agent", "wall", "badBanana", "frozenAgent" };
-        AddVectorObs(rayPer.Perceive(rayDistance, rayAngles, detectableObjects, 0f, 0f));
-        Vector3 localVelocity = transform.InverseTransformDirection(agentRb.velocity);
-        AddVectorObs(localVelocity.x);
-        AddVectorObs(localVelocity.z);
-        AddVectorObs(System.Convert.ToInt32(frozen));
-        AddVectorObs(System.Convert.ToInt32(shoot));
+        if (!useVisual)
```

Comment:
I think this variable should be called `useVectorObs` instead of `useVisual` as if it is not checked and there is a camera on the agent, both visual and vector will be used

Reply:
I agree. Will make the change.